tokio/src/coop.rs

```diff
@@ -151,15 +151,46 @@ cfg_blocking_impl! {
 cfg_coop! {
     use std::task::{Context, Poll};
 
+    #[must_use]
+    pub(crate) struct RestoreOnPending(Cell<Option<Budget>>);
+
+    impl RestoreOnPending {
+        pub(crate) fn made_progress(&self) {
```

Comment:
I don't know if it matters or not, but this could take `self` then do `mem::forget(self)` instead of tracking `Cell<Option<...>>`

Reply:
I specifically chose to use `&self` here so that it's a little less annoying to call `made_progress` if there are closures and such involved. Taking `self` would make the borrow checker much more of a pain, and can't be fixed by `Copy` since implementing `Copy` would make this not work correctly.